autogen/oai/client.py

```diff
@@ -105,6 +105,23 @@ def get_usage(response: ModelClientResponseProtocol) -> Dict:
         """Return usage summary of the response using RESPONSE_USAGE_KEYS."""
         ...  # pragma: no cover
 
+    def update_model(
```

Comment:
I think we are just started this tool and future development is expected. The interface may change. For minimal disruption to the core library, can we create a "subclass" of the protocol in `sample/tools/finetuning` and also bring the implementation of `update_model` method into a subclass of the `OpenAIWrapper`. 

I think there will be a number of future refactors so it's better to keep the `OpenAIWrapper` class lean for now and not take on too many responsibilities.

Reply:
I can subclass the protocol but not `OpenAIWrapper`, that would require conversable agent to be able to hook up this new `OpenAIWrapper` subclass which goes outside of the scope of this feature.

What I can do though is further violate encapsulation inside of the update_model function in the finetune dir

Reply:
> What I can do though is further violate encapsulation inside of the update_model function in the finetune dir

Let's do that. As we continue working on this tool it may be that the encapsulation needs to be updated to accommodate the violation.

Reply:
updated